### DISCUSSION TITLE
How to code the following?

### DISCUSSION THREAD

**Discussion by StarKnight7** [cvxpy/cvxpy#1722] [Q&A] [answered]
I have the following equation that i want to solve,
x*C -E -y = 0

Here, x and y are given; C and E are variables,
x’s size = (128,1),
C’s size = (1,2),
E’s size = (128,2),
y’s size = (128,2),

An important constraint other than the equation mentioned at first is that every row in E has to have the same value. That is E[0][0] = E[1][0] = E[2][0] = … = E[127][0] and
E[0][1] = E[2][1] = … = E[127][1]

How to frame the syntax for this in cvxpy ?

**Comment by rileyjmurray** [comment, marked as answer]
You should define ``e_var = cp.Variable(shape=(1, 2))``, ``ones_vec = np.ones((128,1))``, and ``E = ones_vec @ e_var``. That will ensure that every row of the matrix ``E.value`` is the same.

**Reply by StarKnight7** [reply]
Thank you for the solution :) 

I have one more doubt,... 
I have the variable b, c = cp.Variable(shape = (30,2)) and E(the same E in your comment) and I want to set the constraint as cp.Minimize(cp.sum_squares(A*E - b +numpy.square(c)) . That I want to take the elementwise square of c and use it in the constraint. But its not working. Please help! how to use squares of variables in constraint, especially when it is element wise ?

(cp = cvxpy)
Also, m = 30
n = 20
np.random.seed(1)
A = np.random.randn(m, n)
b = np.random.randn(m,2)
e_var = cp.Variable(shape=(1, 2))
ones_vec = np.ones((20,1))
B = cp.Variable(shape = (30,2))
c = cp.Variable(shape = (30,2))
E = ones_vec @ e_var
